includes/providers/gam/api/class-api.php

```diff
@@ -207,17 +207,34 @@ public function get_session() {
 				'applicationName' => self::APP,
 			],
 		];
-		/** If a network code is not yet available, use first from list. */
-		if ( ! $this->network_code ) {
-			$session  = ( new AdManagerSessionBuilder() )->from( new Configuration( $config ) )->withOAuth2Credential( $this->credentials )->build();
-			$networks = $this->get_networks( $session );
-			if ( ! empty( $networks ) ) {
-				$this->network_code = $networks[0]->getNetworkCode();
+
+		// Get network code and add it to the session config.
+		$config['AD_MANAGER']['networkCode'] = $this->get_network_code( ( new AdManagerSessionBuilder() )->from( new Configuration( $config ) )->withOAuth2Credential( $this->credentials )->build() );
+
+		// Generate and return session.
+		$this->session = ( new AdManagerSessionBuilder() )->from( new Configuration( $config ) )->withOAuth2Credential( $this->credentials )->build();
+		return $this->session;
+	}
+
+	/**
+	 * Whether the given network code is valid for the authenticated user.
+	 *
+	 * @param int $network_code Network code to validate.
+	 *
+	 * @return bool Whether the network code is valid.
+	 */
+	public function validate_network_code( $network_code ) {
```

Comment:
Am I missing something? I don't see this method being invoked anywhere.

Reply:
Whoops! Thank you for catching that, it's residual code from a previous strategy I was going for. Removed in 7599f0741a1bf33bdf6963c9c4434d33b282cc05